app/src/main/java/com/example/movies/ui/watch_trailer/adapter/ViewPagerThumbnailsAdapter.kt

```diff
@@ -0,0 +1,57 @@
+package com.example.movies.ui.watch_trailer.adapter
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import android.widget.ImageView
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.Glide
+import com.example.domain.models.TrailerModel
+import com.example.movies.databinding.CellThumbnailBinding
+
+class ViewPagerThumbnailsAdapter(var onItemClickListener: ((videoKey: String) -> Unit)? = null) :
+    ListAdapter<TrailerModel, ViewPagerThumbnailsAdapter.ThumbnailsViewHolder>(
+        ViewPagerThumbnailsDiffCallback()
+    ) {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ThumbnailsViewHolder {
+        val binding =
+            CellThumbnailBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+        return ThumbnailsViewHolder(binding,parent.context, onItemClickListener)
+
+    }
+
+    override fun onBindViewHolder(holder: ThumbnailsViewHolder, position: Int) {
+        val trailerModel = getItem(position)
+
+        holder.bind(trailerModel)
+        holder.onClickItem(trailerModel.videoKey)
```

Comment:
onClickItem method can be called inside bind. 
Ideally, an adapter calls only holder.bind method. All the rest are done by holder, inside.